app/src/main/java/com/hedvig/app/feature/addressautocompletion/activityresult/FetchDanishAddressContract.kt

```diff
@@ -0,0 +1,43 @@
+package com.hedvig.app.feature.addressautocompletion.activityresult
+
+import android.content.Context
+import android.content.Intent
+import android.os.Parcelable
+import androidx.activity.result.contract.ActivityResultContract
+import com.hedvig.app.feature.addressautocompletion.model.DanishAddress
+import com.hedvig.app.feature.addressautocompletion.ui.AddressAutoCompleteActivity
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Input: DanishAddress? The initial address that will be used as the initial text for autocompletion.
+ * Output: FetchDanishAddressContractResult The result potentially containing the autocomplete address result.
+ */
+class FetchDanishAddressContract : ActivityResultContract<DanishAddress?, FetchDanishAddressContractResult>() {
+    override fun createIntent(context: Context, input: DanishAddress?): Intent {
+        return AddressAutoCompleteActivity.newInstance(context, input)
```

Comment:
The contract itself knows how to start the activity it needs to start for its result